pageserver/src/tenant/ephemeral_file/zero_padded_read_write/zero_padded_buffer.rs

```diff
@@ -0,0 +1,98 @@
+use std::mem::MaybeUninit;
+
+pub struct Buf<const N: usize> {
```

Comment:
nit: perhaps call it `ZeroPaddedBuf`?

Reply:
Hm, I tend to do qualified use because it makes the module implementation less wordy.

In order to justify this, I searched for RFCs :P found https://rust-lang.github.io/rfcs/0356-no-module-prefixes.html

Let's ask style master @koivunej about it :)

Reply:
`zero_padded::Buf` makes more sense; I wouldn't even repeat the `_buffer` because that's an eclipse-autocomplete-java style in my mind; no one would write that by hand. :)

I wasn't aware of the RFC directly, but `std` is an example of this style. Unsure if it made it to rust API guidelines... Cannot see it in [Naming](https://rust-lang.github.io/api-guidelines/naming.html).

Reply:
`Buf` name is a collision with [`bytes::Buf`](https://docs.rs/bytes/latest/bytes/index.html#reexport.Buf) but does it matter, probably not, esp. if the module prefix is used.

Reply:
Happy with `Buf` then. Feel free to resolve.